pep-0646.rst

```diff
@@ -1015,21 +1015,57 @@ Where:
 
     star_annotation: ':' star_expression
 
-This accomplishes the desired outcome (making ``*args: *Ts`` not be a syntax
-error) while matching the behaviour of star-unpacking in other contexts:
-at runtime, ``__iter__`` is called on the starred object, and a tuple
-containing the items of the resulting iterator is set as the type annotion
-for ``args``. In other words, at runtime ``*args: *foo`` is equivalent to
-``*args: tuple(foo)``.
+We also need to deal with the ``star_expression`` that results from this
+construction. Normally, a ``star_expression`` occurs within the context
+of e.g. a list, so a ``star_expression`` is handled by essentially
+calling ``iter()`` on the starred object, and inserting the results
+of the resulting iterator into the list at the appropriate place. For
+``*args: *Ts``, however, we must process the ``star_expression`` in a
+different way.
+
+We do this by instead making a special case for the ``star_expression``
+resulting from ``*args: *Ts``, modifying the compiler to emit bytecode
+corresponding to ``next(iter(Ts))``. While slightly inconsistent with
```

Comment:
I think we shouldn't mention bytecode in a PEP since not all Python implementations use bytecode. But we can mention that it emits code equivalent to
```py
[a] = [*Ts]
```

Reply:
Good point. Done.